### PR TITLE
January 23 2024 - adding Ulli Hafner's spotlight

### PR DESCRIPTION
**Pull request by kmartens27**
This PR is to add the spotlight page for @uhafner. 

I will add the change to the previous contributor spotlight (feature from `true` to `false`) once it is available.

### CODE REVIEW

**Review by lemeurherve — COMMENTED** [review]
Small typo, LGTM otherwise

File: src/pages/contributors/ulli-hafner.adoc [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+He is also part of the UX SIG, participating in meetings and providing various UX contributions over the years.
+Ulli likes to help new contributors grapple with Jenkins and give talks at Jenkins events where he can offer his insight into open source.
+In his free time, Ulli likes to balance his screen-usage by spending time with his family, being outdoors, and being active through sports or swimming.
+
+== What is your background prior to contributing to Jenkins?
+
+I was a software developer and architect, and had never contributed to open source before Jenkins. 
+
+Currently, I am the maintainer of several Jenkins plugins, mostly in the area of build result visualization using modern JS based user interfaces.
+For example, I am the maintainer of the plugin:warnings-ng[Warnings], plugin:code-coverage-api[Code Coverage] and plugin:git-forensics[Git Forensics] plugins, as well as several UI related JS wrapper plugins like plugin:bootsrap5-api[Bootstrap], plugin:prism-api[Prism API], or plugin:echarts-api[ECharts].
```
[the remark this answers is not in the document]
Comment (kmartens27) [reply src/pages/contributors/ulli-hafner.adoc:24]: thanks for pointing that out @lemeurherve !

File: src/pages/contributors/ulli-hafner.adoc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+I am a lifeguard and swimming trainer for children in our junior lifeguard group.
+
+== How long have you been using Jenkins?
+
+I have been using Jenkins and contributing to Jenkins for more than 15 years now.
+
+== Why choose Jenkins over other projects?
+
+I needed Jenkins in one of my company projects.
+I was the architect in the team and one of my responsibilities was to make sure that we deliver only high quality code.
```
Comment (jsmith-cb) [inline src/pages/contributors/ulli-hafner.adoc:41]:
```suggestion
I was the architect in the team, and one of my responsibilities was to make sure that we deliver only high quality code. 
```

File: src/pages/contributors/ulli-hafner.adoc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Currently, I am the maintainer of several Jenkins plugins, mostly in the area of build result visualization using modern JS based user interfaces.
+For example, I am the maintainer of the link:https://plugins.jenkins.io/warnings-ng/[Warnings], link:https://plugins.jenkins.io/code-coverage-api/[Code Coverage] and link:https://plugins.jenkins.io/git-forensics/[Git Forensics] plugins, as well as several UI related JS wrapper plugins like link:https://plugins.jenkins.io/bootstrap5-api/[Bootstrap], link:https://plugins.jenkins.io/prism-api/[Prism API], or link:https://plugins.jenkins.io/echarts-api/[ECharts].
+I am a member of the Jenkins UX SIG and the Jenkins Governance Board.
+
+During the day, I am a professor of Software Engineering at the University of Applied Sciences Munich, where I also try to win new Jenkins contributors by letting students develop new features in their student projects and theses.
+I am also helping new contributors and students in our Jenkins forums or by giving some talks at our Jenkins Meetups. 
+
+I am a father of two children and try to spend time with my family as much as possible.
```
Comment (jsmith-cb) [inline src/pages/contributors/ulli-hafner.adoc:30]:
```suggestion
I am a father of two children, and try to spend time with my family as much as possible.
```

File: src/pages/contributors/ulli-hafner.adoc [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+He is also part of the UX SIG, participating in meetings and providing various UX contributions over the years.
+Ulli likes to help new contributors grapple with Jenkins and give talks at Jenkins events where he can offer his insight into open source.
+In his free time, Ulli likes to balance his screen-usage by spending time with his family, being outdoors, and being active through sports or swimming.
+
+== What is your background prior to contributing to Jenkins?
+
+I was a software developer and architect, and had never contributed to open source before Jenkins. 
+
+Currently, I am the maintainer of several Jenkins plugins, mostly in the area of build result visualization using modern JS based user interfaces.
+For example, I am the maintainer of the link:https://plugins.jenkins.io/warnings-ng/[Warnings], link:https://plugins.jenkins.io/code-coverage-api/[Code Coverage] and link:https://plugins.jenkins.io/git-forensics/[Git Forensics] plugins, as well as several UI related JS wrapper plugins like link:https://plugins.jenkins.io/bootstrap5-api/[Bootstrap], link:https://plugins.jenkins.io/prism-api/[Prism API], or link:https://plugins.jenkins.io/echarts-api/[ECharts].
```
Comment (jsmith-cb) [inline src/pages/contributors/ulli-hafner.adoc:24]:
```suggestion
For example, I am the maintainer of the link:https://plugins.jenkins.io/warnings-ng/[Warnings], link:https://plugins.jenkins.io/code-coverage-api/[Code Coverage], and link:https://plugins.jenkins.io/git-forensics/[Git Forensics] plugins, as well as several UI related JS wrapper plugins like link:https://plugins.jenkins.io/bootstrap5-api/[Bootstrap], link:https://plugins.jenkins.io/prism-api/[Prism API], or link:https://plugins.jenkins.io/echarts-api/[ECharts].
```

File: src/pages/contributors/ulli-hafner.adoc [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+:page-image: avatar/ulli-hafner.png
+:page-pronouns: He/Him/His
+:page-location: Munich, Bavaria, Germany
+:page-firstcommit: 2007
+:page-datepublished: 2024-01-10
+:page-featured: true
+:page-intro: Ulli Hafner is a professor of Software Engineering at the University of Applied Sciences Munich where he advocates for Jenkins in several ways. He is also a software developer, open-source contributor, plugin maintainer and serves on the Jenkins governance board.
+
+He is also part of the UX SIG, participating in meetings and providing various UX contributions over the years.
+Ulli likes to help new contributors grapple with Jenkins and give talks at Jenkins events where he can offer his insight into open source.
```
Comment (jsmith-cb) [inline src/pages/contributors/ulli-hafner.adoc:16]:
```suggestion
Ulli likes to help new contributors grapple with Jenkins and give talks at Jenkins events, where he can offer his insight into open source.
```

File: src/pages/contributors/ulli-hafner.adoc [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+:page-twitter: 
+:page-github: uhafner
+:page-email: 
+:page-image: avatar/ulli-hafner.png
+:page-pronouns: He/Him/His
+:page-location: Munich, Bavaria, Germany
+:page-firstcommit: 2007
+:page-datepublished: 2024-01-10
+:page-featured: true
+:page-intro: Ulli Hafner is a professor of Software Engineering at the University of Applied Sciences Munich where he advocates for Jenkins in several ways. He is also a software developer, open-source contributor, plugin maintainer and serves on the Jenkins governance board.
```
Comment (jsmith-cb) [inline src/pages/contributors/ulli-hafner.adoc:13]:
```suggestion
:page-intro: Ulli Hafner is a professor of Software Engineering at the University of Applied Sciences Munich where he advocates for Jenkins in several ways. He is also a software developer, open-source contributor, plugin maintainer, and serves on the Jenkins governance board.
```

**Review by jsmith-cb — APPROVED** [review]
Few punctuation suggestions

**Comment by kmartens27** [comment]
The publication date has shifted two weeks, to align with the updated publication schedule & short break at the end of the year for Jenkins.

**Review by krisstern — APPROVED** [review]
LGTM

**Comment by kmartens27** [comment]
@uhafner when you have a moment, would you be able to review and share if there is any feedback or concerns prior to publishing? Things should be fairly the same from our original draft, but just want to make sure everything looks good!

**Comment by kmartens27** [comment]
I have adjusted the publication date so that Ulli's spotlight is published tomorrow (January 23) instead of the 24th, due to the LTS release. With the release, there may be follow up questions or news, and it would be better to publish Ulli's spotlight prior, so that it does not get lost in the other news.

**Review by lemeurherve — APPROVED** [review]

File: src/pages/contributors/ulli-hafner.adoc [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+Ulli likes to help new contributors grapple with Jenkins and give talks at Jenkins events, where he can offer his insight into open source.
+In his free time, Ulli likes to balance his screen-usage by spending time with his family, being outdoors, and being active through sports or swimming.
+
+== What is your background prior to contributing to Jenkins and outside of Jenkins? 
+
+I was a software developer and architect, and had never contributed to open source before Jenkins. 
+
+Currently, I am the maintainer of several Jenkins plugins, mostly in the area of build result visualization using modern JS based user interfaces.
+For example, I am the maintainer of the link:https://plugins.jenkins.io/warnings-ng/[Warnings], link:https://plugins.jenkins.io/code-coverage-api/[Code Coverage], and link:https://plugins.jenkins.io/git-forensics/[Git Forensics] plugins, as well as several UI related JS wrapper plugins like link:https://plugins.jenkins.io/bootstrap5-api/[Bootstrap], link:https://plugins.jenkins.io/prism-api/[Prism API], or link:https://plugins.jenkins.io/echarts-api/[ECharts].
```
Comment (uhafner) [inline src/pages/contributors/ulli-hafner.adoc:24]: The plugin coordinates changed last month:

```suggestion
For example, I am the maintainer of the link:https://plugins.jenkins.io/warnings-ng/[Warnings], link:https://plugins.jenkins.io/coverage/[Coverage], and link:https://plugins.jenkins.io/git-forensics/[Git Forensics] plugins, as well as several UI-related JS-wrapper plugins like link:https://plugins.jenkins.io/bootstrap5-api/[Bootstrap], link:https://plugins.jenkins.io/prism-api/[Prism API], or link:https://plugins.jenkins.io/echarts-api/[ECharts].
```

File: src/pages/contributors/ulli-hafner.adoc [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+== What sort of contributions have felt the most successful or impactful?
+
+I think the most impactful contributions are the plugins for static analysis and code coverage, in combination with the git forensics plugin.
+Both plugins have a user base of more than 25,000 installations.
+The static analysis plugins won the first prize of the SUN Open Source Community Innovation Awards Program in 2008.
+
+== Advice for new developers and new members of the open-source community
+
+Do not reinvent the wheel and keep, or improve, the quality standards of the existing projects.
```
Comment (uhafner) [inline src/pages/contributors/ulli-hafner.adoc:61]:
```suggestion
Do not reinvent the wheel and keep, or improve the quality standards of the existing projects.
```

File: src/pages/contributors/ulli-hafner.adoc [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+I have been using Jenkins and contributing to Jenkins for more than 15 years now.
+
+== Why choose Jenkins over other projects?
+
+I needed Jenkins in one of my company projects.
+I was the architect in the team, and one of my responsibilities was to make sure that we deliver only high quality code. 
+
+== What problems has Jenkins solved for you?
+
+Jenkins helped me ensure that a lot of quality gates, that the team defined in the definition of "done", could be automatically checked and reported in a rich user interface.
```
Comment (uhafner) [inline src/pages/contributors/ulli-hafner.adoc:45]:
```suggestion
Jenkins helped me ensure that a lot of quality gates, that the team defined in the "Definition of Done", could be automatically checked and reported in a rich user interface.  
```

File: src/pages/contributors/ulli-hafner.adoc [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+:page-image: avatar/ulli-hafner.png
+:page-pronouns: He/Him/His
+:page-location: Munich, Bavaria, Germany
+:page-firstcommit: 2007
+:page-datepublished: 2024-01-23
+:page-featured: true
+:page-intro: Ulli Hafner is a professor of Software Engineering at the University of Applied Sciences Munich where he advocates for Jenkins in several ways. He is also a software developer, open-source contributor, plugin maintainer, and serves on the Jenkins governance board.
+
+He is also part of the UX SIG, participating in meetings and providing various UX contributions over the years.
+Ulli likes to help new contributors grapple with Jenkins and give talks at Jenkins events, where he can offer his insight into open source.
```
Comment (uhafner) [inline src/pages/contributors/ulli-hafner.adoc:16]: Shouldn't this be either 
- "and gives talks at Jenkins events"
- or "and likes to give talks at Jenkins events"?

File: src/pages/contributors/ulli-hafner.adoc [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+I am a lifeguard and swimming trainer for children in our junior lifeguard group.
+
+== How long have you been using Jenkins?
+
+I have been using Jenkins and contributing to Jenkins for more than 15 years now.
+
+== Why choose Jenkins over other projects?
+
+I needed Jenkins in one of my company projects.
+I was the architect in the team, and one of my responsibilities was to make sure that we deliver only high quality code.
```
Comment (uhafner) [inline src/pages/contributors/ulli-hafner.adoc:41]:
```suggestion
I was the architect in the team, and one of my responsibilities was to make sure that we deliver only high-quality code. 
```

**Review by uhafner — APPROVED** [review]
I added some small refinements that where highlighted by Grammarly, so feel free to apply if they make sense to you as well.

File: src/pages/contributors/ulli-hafner.adoc (hunk 10 above)
Comment (lemeurherve) [reply src/pages/contributors/ulli-hafner.adoc:16]: It looks like to me that a "to" is implied here:
```suggestion
Ulli likes to help new contributors grapple with Jenkins and to give talks at Jenkins events, where he can offer his insight into open source.
```

Comment (kmartens27) [reply src/pages/contributors/ulli-hafner.adoc:16]: I changed the verbiage to more active voicing.
```suggestion
Ulli enjoys helping new contributors grapple with Jenkins and giving talks at Jenkins events, where he can offer his insight into open source.
```

**Comment by kmartens27** [comment]
Thanks very much @uhafner for your review and feedback, I've updated everything accordingly!

**Review by lemeurherve — APPROVED** [review]

**Pull request merged**